src/Tizen.NUI/src/public/BaseComponents/ControlState.cs

```diff
@@ -0,0 +1,280 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+
+namespace Tizen.NUI.BaseComponents
+{
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public class ControlState : IEquatable<ControlState>, IComparable<ControlState>
+    {
+        private static readonly Dictionary<string, ControlState> stateDictionary = new Dictionary<string, ControlState>();
+        //Default States
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public static readonly ControlState All = Create("All");
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public static readonly ControlState Normal = Create("Normal");
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public static readonly ControlState Focused = Create("Focused");
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public static readonly ControlState Pressed = Create("Pressed");
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public static readonly ControlState Disabled = Create("Disabled");
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public static readonly ControlState Selected = Create("Selected");
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public static readonly ControlState DisabledSelected = Disabled + Selected;
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public static readonly ControlState DisabledFocused = Disabled + Focused;
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public static readonly ControlState SelectedFocused = Selected + Focused;
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public static readonly ControlState Other = Create("Other");
+
+        private static readonly char[] delimiterChars = { ' ', ',', ':', '|' };
+
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public bool IsCombined => stateSet.Count > 1;
+
+        private readonly List<ControlState> stateSet;
+        private readonly string name;
+
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public ControlState() => stateSet = new List<ControlState>();
+
+        private ControlState(string name) : this() => this.name = name;
+
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public static ControlState Create(string name)
+        {
+            if (stateDictionary.TryGetValue(name, out ControlState state))
```

Comment:
Is state name case sensitive?

`All` and `all`
`Normal` and `normal`
and so on..

Reply:
Yes, state name is case sensitive.